### DISCUSSION TITLE
Can we run Data Migration(.ts/.js/.mjs) after a specific migration file?

### DISCUSSION THREAD

**Discussion by mydjolie** [prisma/prisma#22772] [Q&A]
### Question

Hi !

I'm working on a project which has already launched 🚀 (which already has users.)
We have decided to change schema and need to migrate the data.

What I expected to be done in the same migration file :
- create new` CourseBlackList`
- migrate data from `User.status` to the `CourseBlackList`
- drop the column `status` from `User`

Normally, I can just use `UPDATE "Table" SET ...... ` but this time 
I must use INSERT INTO with the data I queried from another table in order to not lose the data. 
```
INSERT INTO CourseBlackList ("userId", "courseId", "isBlackListed", "reason", "description")
SELECT
  "User"."id" AS userId,
  "_CourseToUser"."A" AS courseId,
  true AS isBlackListed,
  null AS reason,
  null AS description
FROM
  "User"
  INNER JOIN "_CourseToUser" ON "User"."id" = "_CourseToUser"."B"
WHERE
  "User"."status" = 'PAUSED';
```
Then `ALTER TABLE "User" DROP COLUMN "status",`

 and Prisma keep giving this error like Error: P1014 -The underlying table for model `(CourseBlackList)` does not exist.
 but when I deleted the INSERT Part, Prisma run perfectly.
 
 I have found Data Migration in Prisma Website, I created data-migration and be able to execute the file.
 
 The Question is : ** Is there any way to make Prisma run the specific Data Migration script/file before applying the next migration file?**
 
 Another question : Why INSERT INTO doesn't work in this case?

**Comment by ludralph** [comment]
Hi @mydjolie 👋 

Thank you for raising this question.

> Is there any way to make Prisma run the specific Data Migration script/file before applying the next migration file?

Currently there is no way in Prisma to allow running a specific Migration file or script. However, we do have open [feature request](https://github.com/prisma/prisma/issues/14916) to support this functionality. You can add your usecase to the issue and also leave a 👍 to help our engineers prioritize it.

>  Why INSERT INTO doesn't work in this case?

The error message suggest that Prisma is unable to find the table you are trying to insert into. This could be due to a number of reasons, such as the table not being created yet at the time the migration is run, or a mismatch between the table name in the database and the model name in your Prisma schema.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
